climpred_skeleton/alignment.py

```diff
@@ -28,11 +28,45 @@ def __init__(
     ):
         super().__init__(initialized, observation)
         self._alignment = alignment
+        self._all_verifs = self._observation['time'].data
+        self._all_inits = self._initialized['init'].data
+
+        # Run comparison on the object
+        # (this might go to Scoring, since PM doesn't go through here)
         comparison_obj = Comparison(
             self._initialized, self._observation
         ).get_comparison(comparison)
         self._comparison_method = comparison_obj._comparison_method
         self._initialized, self._observation = comparison_obj.broadcast()
 
+    def _construct_init_lead_matrix(self):
+        """Constructs the init-lead matrix to figure out which inits and verif dates
+        to use at a given lead for a given alignment strategy."""
+        n, freq = self._get_all_lead_cftime_shift_args()
+        init_lead_matrix = xr.concat(
+            [
+                xr.DataArray(
+                    self._shift_hindcast_inits(n, freq),
+                    dims=['init'],
+                    coords=[self._all_inits],
+                )
+                for n in n
+            ],
+            'lead',
+        )
+        return init_lead_matrix
+
+    def _shift_hindcast_inits(self, n: int, freq: str):
+        """Shifts hindcast inits `n` timesteps forward at `freq`."""
+        time_index = self._initialized['init'].to_index()
+        return time_index.shift(n, freq)
+
+
+class SameInitializations(LeadAlignment):
+    """Class for `same_inits` keyword."""
+
+    def __init__(self, initialized, observation, comparison, alignment):
```

Comment:
Arguments getting long here and will continue getting longer downstream; my mind is blanking out how to handle this.

Reply:
I thought it was the point of this inheriting that we carry argument less explicit...